javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentStarterImpl.java

```diff
@@ -115,6 +118,29 @@ public void start() {
     }
   }
 
+  private void init() {
+    instrumentInetAddress();
+  }
+
+  private void instrumentInetAddress() {
+    InetAddressClassFileTransformer transformer = new InetAddressClassFileTransformer();
+    instrumentation.addTransformer(transformer, true);
+
+    try {
+      Class<?> clazz = Class.forName("java.net.InetAddress", false, null);
+      if (transformer.transformed) {
+        // InetAddress was loaded and got transformed
+        return;
+      }
+      // InetAddress was already loaded before we set up transformer
+      instrumentation.retransformClasses(clazz);
+    } catch (ClassNotFoundException | UnmodifiableClassException ignore) {
+      // ignore
+    } finally {
+      instrumentation.removeTransformer(transformer);
+    }
```

Comment:
I'd delete this. With your current implementation if the class was loaded before the transformer was added retransform would fail anyway because retransform can not change class shape (add methods).